pkg/reconciler/security/istio/eventpolicybinding/eventpolicybinding.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package eventpolicybinding
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	"knative.dev/pkg/logging"
+	pkgreconciler "knative.dev/pkg/reconciler"
+	"knative.dev/pkg/tracker"
+
+	"github.com/google/knative-gcp/pkg/apis/security/v1alpha1"
+	bindingreconciler "github.com/google/knative-gcp/pkg/client/injection/reconciler/security/v1alpha1/eventpolicybinding"
+	securitylisters "github.com/google/knative-gcp/pkg/client/listers/security/v1alpha1"
+	"github.com/google/knative-gcp/pkg/reconciler"
+	"github.com/google/knative-gcp/pkg/reconciler/security/istio/eventpolicybinding/resources"
+)
+
+// Reconciler reconciles the EventPolicyBinding.
+type Reconciler struct {
+	*reconciler.Base
+
+	eventPolicyLister       securitylisters.EventPolicyLister
+	httpPolicyLister        securitylisters.HTTPPolicyLister
+	httpPolicyBindingLister securitylisters.HTTPPolicyBindingLister
+
+	policyTracker tracker.Interface
+}
+
+var _ bindingreconciler.Interface = (*Reconciler)(nil)
+
+// ReconcileKind reconciles the EventPolicyBinding.
+func (r *Reconciler) ReconcileKind(ctx context.Context, b *v1alpha1.EventPolicyBinding) pkgreconciler.Event {
+	logging.FromContext(ctx).Debug("Reconciling", zap.Any("EventPolicyBinding", b))
+	b.Status.InitializeConditions()
+	b.Status.ObservedGeneration = b.Generation
+
+	p, err := r.eventPolicyLister.EventPolicies(b.Spec.Policy.Namespace).Get(b.Spec.Policy.Name)
+	if err != nil {
+		logging.FromContext(ctx).Error("Problem getting EventPolicy", zap.Any("EventPolicy", b.Spec.Policy), zap.Error(err))
+		b.Status.MarkBindingFailure("GetPolicyFailure", "%v", err)
+		return fmt.Errorf("failed to get EventPolicy: %w", err)
+	}
+	// Track referenced policy.
+	r.policyTracker.TrackReference(tracker.Reference{
```

Comment:
does this return an error? I think it used to, unless they changed the API. If it does, then we should check for the error...
I've seen cases where we don't have proper permissions to track things and this can fail